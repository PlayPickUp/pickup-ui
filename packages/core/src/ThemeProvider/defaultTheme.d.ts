import { DefaultTheme } from "../../../../types";
export declare const defaultTheme: DefaultTheme;
//# sourceMappingURL=defaultTheme.d.ts.map
