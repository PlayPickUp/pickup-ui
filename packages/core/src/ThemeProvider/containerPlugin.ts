/*
 *  Adds container params to all selectors
 *  Example: 'h1' becomes 'body #PickUpUI h1'
 */

import { Plugin } from "jss";

export function containUI(withReset: boolean, container?: string): Plugin {
  const prefix = container ? container : "body #PickUpUI ";

  // 'rule' is left untyped because the Rule import from jss is not recognizing all child properties
  function onProcessRule(rule): Plugin["onProcessRule"] {
    const parent = rule.options.parent;
    if (
      rule.type !== "style" ||
      rule.selectorText.includes(prefix) ||
      (parent && parent.type === "keyframes") ||
      (withReset && parent && parent.key === "@global")
    )
      return;

    if (rule.selectorText.includes(",")) {
      const allSelectors = rule.selectorText.split(",");
      const transformedSelectors = allSelectors.map((selector: string) =>
        selector.includes(prefix) ? selector : prefix + selector
      );
      rule.selectorText = transformedSelectors.join(",");
    } else {
      rule.selectorText = prefix + rule.selectorText;
    }
  }
  return { onProcessRule };
}
