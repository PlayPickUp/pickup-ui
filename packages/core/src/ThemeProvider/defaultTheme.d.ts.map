{"version":3,"file":"defaultTheme.d.ts","sourceRoot":"","sources":["defaultTheme.ts"],"names":[],"mappings":"AAAA,OAAO,EAMN,YAAY,EAWZ,MAAM,mBAAmB,CAAC;AAqJ3B,eAAO,MAAM,YAAY,EAAE,YAU1B,CAAC"}