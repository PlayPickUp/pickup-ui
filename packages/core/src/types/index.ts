import { FieldInputProps, FormikErrors, FormikTouched } from "formik";
import React, {
  ChangeEvent,
  CSSProperties,
  MouseEvent,
  ReactElement,
  ReactNode,
  SetStateAction,
  Dispatch,
} from "react";

/**
 *
 * Theme Specific Types
 *
 */
export interface ThemeProviderProps {
  children: any;
  withReset?: boolean;
  theme?: DefaultTheme;
}

export interface PurpleMap {
  light: string;
  base: string;
  dark: string;
}

export interface GreenMap {
  light: string;
  base: string;
}

export interface GreyMap {
  light: string;
  base: string;
  dark: string;
}

export interface RedMap {
  light: string;
  base: string;
}

export interface DefaultThemeColors {
  primary: PurpleMap;
  secondary: GreyMap;
  green: GreenMap;
  purple: PurpleMap;
  black: string;
  white: string;
  success: string;
  grey: GreyMap;
  red: RedMap;
  error: string;
}

export interface FontFamilies {
  body: string;
  headline: string;
}

interface FontProperties {
  fontFamily: string;
  fontWeight: number | string;
  fontStyle?: string;
  fontSize: number | string;
  lineHeight: string;
  letterSpacing: string;
  textTransform?: string;
  color: string;
}

export interface FontStyles {
  title: FontProperties;
  heading2: FontProperties;
  heading3: FontProperties;
  body: FontProperties;
  body2: FontProperties;
}

export interface ResponsiveFonts {
  desktop: FontStyles;
  mobile: FontStyles;
}

export interface DefaultTypography {
  fontFamilies: FontFamilies;
  fontStyles: ResponsiveFonts;
}
export type CreateShadow = (
  a: number,
  b: number,
  c: number,
  d: number,
  e: number,
  f: number,
  g: number,
  h: number,
  i: number,
  j: number,
  k: number,
  l: number
) => string;

export type DefaultSpacing = {
  base: number;
};

export type DefaultShadows = Array<string>;

export interface DefaultZIndex {
  overlay: number;
}

export interface DefaultGradients {
  main: string;
  hover: string;
}

export type MediaQuery = (minWidth: number) => string;

export interface Breakpoints {
  small: number;
  medium: number;
  large: number;
}

export type DefaultTheme = {
  borderRadius: number;
  breakpoints: Breakpoints;
  mediaQuery: MediaQuery;
  colors: DefaultThemeColors;
  gradients: DefaultGradients;
  shadows: DefaultShadows;
  spacing: DefaultSpacing;
  typography: DefaultTypography;
  zIndex: DefaultZIndex;
};

/**
 *
 * Button Component
 *
 */

export type ButtonVariant = "fit" | "full";

export interface ButtonProps {
  children?: ReactNode;
  disabled?: boolean;
  style?: React.CSSProperties;
  variant?: ButtonVariant;
  onClick?: (e?: React.MouseEvent) => unknown;
  element?: keyof JSX.IntrinsicElements;
  to?: string;
  href?: string;
  className?: string;
  useSubmit?: boolean;
  submitText?: string;
  color?: "primary" | "secondary" | "light";
}

/**
 *
 * Typography Component
 *
 */

export interface TypographyProps {
  children: ReactNode;
  variant?:
    | "title"
    | "body"
    | "body2"
    | "heading2"
    | "heading3"
    | "heading4"
    | "heading5"
    | "heading6"
    | "span";
  className?: string;
  element?: "p" | "span" | "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
  style?: CSSProperties;
  color?: string;
  useUnescape?: boolean;
}

export type TypographyElementMap = Record<string, keyof JSX.IntrinsicElements>;

/**
 *
 * Fab (Floating Action Button) Component
 *
 */

export interface FabProps {
  className?: string;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  style?: React.CSSProperties;
  icon: any;
  title: string;
  size?: "small" | "large";
  onClick: (e: MouseEvent) => unknown;
  disabled?: boolean;
  disablePopOver?: boolean;
  color?: string;
  iconColor?: string;
}

/**
 *
 * Table Component
 *
 */
export interface HeadCell {
  disablePadding: boolean;
  id: keyof TableData;
  label: string;
  numeric?: boolean;
  width?: number | undefined;
}

export type TableData = Record<any, any>;

export type TableOrder = "asc" | "desc";
export interface DataTableProps {
  headCells: HeadCell[];
  rows: Array<Record<string | number, any>>;
  defaultSortColumn: string;
  tableTitle: string;
  actionToolbar: (props: any) => ReactElement;
}

export interface EnhancedTableProps {
  classes: ReturnType<any>;
  numSelected: number;
  onRequestSort: (event: React.MouseEvent<unknown>, property: any) => void;
  onSelectAllClick: (event: React.ChangeEvent<HTMLInputElement>) => void;
  order: TableOrder;
  orderBy: string;
  rowCount: number;
  headCells: HeadCell[];
}

export interface EnhancedTableToolbarProps {
  numSelected: number;
  tableTitle: string;
  actionToolbar: (props: any) => ReactElement;
  selected: string[];
  setSelected: Dispatch<SetStateAction<string[]>>;
}

/**
 *
 * Icon Component
 *
 */
export interface IconBaseProps {
  children: ReactElement;
  color?: string;
  size?: number;
  className?: string;
  style?: React.CSSProperties;
}

export interface BaseFormikFields {
  id: string;
  name: string;
  field: FieldInputProps<any>;
  form: Record<any, any>;
}

export type BaseFormikHandleChange = (e: ChangeEvent) => unknown;

export interface BaseFormikFieldsWithHandleChange extends BaseFormikFields {
  handleChange: BaseFormikHandleChange;
}

/**
 *
 * TextInput Component
 *
 */
export interface TextInputProps extends BaseFormikFieldsWithHandleChange {
  className?: string;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  style?: React.CSSProperties;
  placeholder?: string;
  label?: string;
}

/**
 *
 * Label Component
 *
 */
export interface LabelProps {
  htmlFor: string;
  children: string | ReactNode;
  className?: string;
  style?: React.CSSProperties;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
}

/**
 *
 * TextArea Component
 *
 */
export interface TextAreaProps extends BaseFormikFieldsWithHandleChange {
  className?: string;
  style?: React.CSSProperties;
  innerClassName?: string;
  innerStyle?: React.CSSProperties;
  label?: string;
}

/**
 *
 * Select Component
 *
 */ export interface SelectItem {
  label: string;
  value: string;
}

export type BaseFormikSetFieldValue = (
  field: string,
  value: any,
  shouldValidate?: boolean | undefined
) => unknown;

export interface SelectProps extends BaseFormikFields {
  className?: string;
  items: SelectItem[];
  label: string;
  placeholder?: string;
  style?: React.CSSProperties;
}

export interface DropdownStyleProps {
  isOpen: boolean;
}

/**
 *
 * NestedInput Component
 *
 */
export interface NestedInputProps extends BaseFormikFieldsWithHandleChange {
  buttonText: string;
  placeholder: string;
}

/**
 *
 * MultiSelect Component
 *
 */
export interface MultiSelectProps extends BaseFormikFields {
  items: SelectItem[];
  label: string;
}

/**
 *
 * Individual Icon Component
 *
 */
export interface IconProps {
  color?: string;
  className?: string;
  style?: React.CSSProperties;
}

/**
 *
 * Chip Component
 *
 */
export interface ChipProps {
  disabled?: boolean;
  onClick?: (e: React.MouseEvent) => unknown;
  color?: "default" | "primary" | "secondary";
  className?: string;
  style?: React.CSSProperties;
  isActive?: boolean;
  label: string;
  element?: keyof JSX.IntrinsicElements;
  href?: string;
}

/**
 *
 * Paper Component
 *
 */
export interface PaperProps {
  padding?: number;
  children: unknown;
  style?: React.CSSProperties;
  className?: string;
}

/**
 *
 * FormError Component
 *
 */
export interface FormErrorProps {
  errors: FormikErrors<unknown>;
  touched: FormikTouched<unknown>;
  name: string;
}

/**
 *
 * Loader Component
 *
 */
export interface LoaderProps {
  className?: string;
  style?: React.CSSProperties;
  delay?: number;
  text?: string;
}

/**
 *
 * Picker Button
 *
 */
export interface PickerButtonProps {
  className?: string;
  displayText: string;
  onClick: React.MouseEventHandler<HTMLButtonElement>;
  style?: React.CSSProperties;
  showResult?: boolean;
  result: number | null;
  isPick: boolean;
}
