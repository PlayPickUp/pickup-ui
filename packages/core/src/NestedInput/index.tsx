import React, { useState, useRef } from "react";
import { createUseStyles } from "react-jss";

import Button from "../Button";

import { DefaultTheme, NestedInputProps } from "../types";
import FormError from "../FormError";

import "react-phone-number-input/style.css";
import PhoneInput from "react-phone-number-input/input";

const useStyles = createUseStyles((theme: DefaultTheme) => ({
  root: {
    display: "flex",
    position: "relative",
    flexFlow: "row nowrap",
    justifyContent: "space-between",
    alignItems: "center",
    border: (props) =>
      props.inputFocus
        ? `1px solid ${theme.colors.primary.base}`
        : `1px solid ${theme.colors.grey.medium}`,
    borderRadius: theme.borderRadius * 2,
    backgroundColor: theme.colors.white,
    width: "100%",
    maxWidth: 360,
    height: (props) => (props.usePhoneNumber ? 60 : 56),
    padding: [0, theme.spacing.base * 1.5],
  },
  inputContainer: {
    flex: "1 1 auto",
    paddingRight: 8,
  },
  input: {
    height: (props) =>
      props.usePhoneNumber || props.useVerificationCode ? 22 : 48,
    width: "100%",
    fontFamily: theme.typography.fontFamilies.body,
    fontSize: 16,
    lineHeight: (props) =>
      props.usePhoneNumber || props.useVerificationCode ? "22px" : "24px",
    letterSpacing: "0.1px",
    color: theme.colors.grey.dark,
    appearance: "none",
    border: "none",
    "&:focus": {
      border: "none",
      outline: "none",
      "& $root": {
        borderColor: theme.colors.grey.dark,
      },
    },
    "&:disabled": {
      backgroundColor: "transparent",
    },
  },
  phoneContainer: {
    display: "flex",
    position: "relative",
    flexFlow: "row nowrap",
    alignItems: "center",
  },
  phonePrefix: {
    position: "relative",
    fontSize: 15,
    lineHeight: "22px",
    fontFamily: theme.typography.fontFamilies.body,
    color: theme.colors.grey.dark,
    padding: [0, theme.spacing.base * 2],
  },
  phoneLabel: {
    fontSize: 11,
    lineHeight: "18px",
    fontFamily: theme.typography.fontFamilies.body,
    color: theme.colors.grey.base,
    position: "relative",
    display: "block",
    top: -3,
    left: 5,
  },
  phoneButtonSmall: {
    fontSize: 14,
    padding: [0, theme.spacing.base * 2],
    margin: 0,
    fontWeight: "normal",
  },
  breakpointMini: {
    display: "inline",
    [theme.mediaQuery(320)]: {
      display: "none",
    },
  },
  breakpointLarge: {
    display: "none",
    [theme.mediaQuery(320)]: {
      display: "inline",
    },
  },
}));

const NestedInput: React.FC<NestedInputProps> = ({
  buttonText,
  placeholder,
  label,
  usePhoneNumber,
  useVerificationCode,
  disabled,
  useSubmit = true,
  onClick,
  ...props
}) => {
  const [inputFocus, setInputFocus] = useState<boolean>(false);
  const classes = useStyles({
    inputFocus,
    usePhoneNumber,
    useVerificationCode,
  });
  const inputRef = useRef();

  const toggleFocus = () => setInputFocus(!inputFocus);
  const handleFocus = (el: HTMLInputElement): void => {
    if (!el) {
      return;
    }
    el.focus();
    toggleFocus();
  };

  return (
    <>
      <div className={classes.root}>
        <div className={classes.inputContainer}>
          {(usePhoneNumber || useVerificationCode) && label ? (
            <label
              className={classes.phoneLabel}
              htmlFor={props.field.name}
              onClick={() => handleFocus(inputRef.current)}
            >
              {label}
            </label>
          ) : null}
          {usePhoneNumber ? (
            <div className={classes.phoneContainer}>
              <div className={classes.phonePrefix}>+1</div>
              <PhoneInput
                {...props.field}
                id={props.id}
                // @ts-ignore
                ref={inputRef}
                className={classes.input}
                country="US"
                placeholder="(345) 555-2343"
                smartCaret={false}
                disabled={disabled}
                onChange={(e) => props.form.setFieldValue(props.field.name, e)}
                onBlur={(e) => {
                  toggleFocus();
                  props.field.onBlur(e);
                }}
                onFocus={toggleFocus}
              />
            </div>
          ) : (
            <input
              {...props.field}
              id={props.id}
              className={classes.input}
              placeholder={placeholder}
              onFocus={toggleFocus}
              onBlur={(e) => {
                toggleFocus();
                props.field.onBlur(e);
              }}
            />
          )}
        </div>
        <div
          className={classes.breakpointLarge}
          data-testid="breakpoint-large-div"
        >
          {useSubmit ? (
            <Button
              style={{ minWidth: 90 }}
              className={
                usePhoneNumber || useVerificationCode
                  ? classes.phoneButtonSmall
                  : null
              }
              useSubmit
              submitText={buttonText}
              disabled={disabled}
            />
          ) : (
            <Button
              style={{ minWidth: 90 }}
              data-testid="pickup-nested-button"
              className={
                usePhoneNumber || useVerificationCode
                  ? classes.phoneButtonSmall
                  : null
              }
              disabled={disabled}
              submitText={buttonText}
              onClick={(e) => {
                e.preventDefault();
                onClick();
              }}
            >
              {buttonText}
            </Button>
          )}
        </div>
      </div>
      <div
        className={classes.breakpointMini}
        data-testid="breakpoint-small-div"
      >
        {useSubmit ? (
          <Button
            style={{ minWidth: 90, marginTop: 4 }}
            className={
              usePhoneNumber || useVerificationCode
                ? classes.phoneButtonSmall
                : null
            }
            useSubmit
            variant="full"
            submitText={buttonText}
            disabled={disabled}
          />
        ) : (
          <Button
            style={{ minWidth: 90, marginTop: 4 }}
            className={
              usePhoneNumber || useVerificationCode
                ? classes.phoneButtonSmall
                : null
            }
            variant="full"
            disabled={disabled}
            submitText={buttonText}
            onClick={(e) => {
              e.preventDefault();
              onClick();
            }}
          >
            {buttonText}
          </Button>
        )}
      </div>
      <FormError
        errors={props.form.errors}
        touched={props.form.touched}
        name={props.field.name}
      />
    </>
  );
};

export default NestedInput;
