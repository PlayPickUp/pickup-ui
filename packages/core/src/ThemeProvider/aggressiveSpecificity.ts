/* This is based on the unsupported jss-increase-specificity plugin at https://github.com/iamstarkov/jss-increase-specificity
JSS's original createGenerateId has been adapted here to use logic from that plugin. /*

// From the original plugin:
/**
 * use :not(#\20), :not(.\20) and :not(\20) instead of generating unlikely
 * appearing ids…
 * — twitter.com/subzey/status/829050478721896448
 * Rationale: \20 is a css escape for U+0020 Space, and neither classname,
 * nor id, nor tagname can contain a space
 * — twitter.com/subzey/status/829051085885153280
 */

import { Rule, CreateGenerateId } from "jss";
import { JssStyleSheet } from "../types";

var selector = ":not(#\\20)";
const repeat = 3;
const maxRules = 1e10;

export const increaseSpecificity: CreateGenerateId = (options = {}) => {
  let ruleCounter = 0;
  console.log({ options });
  return (rule: Rule, sheet?: JssStyleSheet): string => {
    ruleCounter += 1;
    if (ruleCounter > maxRules) {
      console.warn(
        false,
        `[JSS] You might have a memory leak. Rule counter is at ${ruleCounter}.`
      );
    }
    const jssId = String(sheet.options.jss.id) ?? "";
    const prefix = sheet.options.classNamePrefix ?? "";

    let suffix = "";
    for (let i = 0; i < repeat; i++) {
      suffix += selector;
    }

    return `${prefix + rule.key}${
      jssId ? `-${jssId}` : ""
    }-${ruleCounter}${suffix}`;
  };
};
