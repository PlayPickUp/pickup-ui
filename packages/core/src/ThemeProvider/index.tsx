import React from "react";
import { jss, JssProvider, ThemeProvider as JSSThemeProvider } from "react-jss";
import { containUI } from "./containerPlugin";
import merge from "lodash/merge";

import { defaultTheme } from "./defaultTheme";

import { ThemeProviderProps } from "../types";
import GlobalsAndReset from "./GlobalsAndReset";

function generateRandom(digitCount) {
  return Math.floor(
      Math.pow(10, digitCount - 1) +
      Math.random() * 9 * Math.pow(10, digitCount - 1)
  );
}

const ThemeProvider: React.FC<ThemeProviderProps> = ({
  children,
  withReset = true,
  theme,
  ...rest
}) => {
  jss.use(containUI(withReset));
  const mergedTheme = theme ? merge(defaultTheme, theme) : defaultTheme;

  const random = generateRandom(5)

  return (
    <div id="PickUpUI">
      <JssProvider jss={jss} classNamePrefix={`PU${random}--`}>
        <JSSThemeProvider theme={mergedTheme} {...rest}>
          <GlobalsAndReset>{children}</GlobalsAndReset>
        </JSSThemeProvider>
      </JssProvider>
    </div>
  );
};

export default ThemeProvider;
