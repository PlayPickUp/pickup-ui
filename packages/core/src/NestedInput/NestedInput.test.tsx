import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import jssSerializer from "jss-snapshot-serializer";
expect.addSnapshotSerializer(jssSerializer);
import { Formik, Form, Field } from "formik";
import ThemeProvider from "../ThemeProvider";
import NestedInput from "./index";
import { NestedInputProps } from "../types";

const sleep = (ms) => new Promise((r) => setTimeout(r, ms));
interface OptionalProps extends NestedInputProps {
  onSubmit?: jest.Mock;
}
const NestedInputFormik = (
  optionalProps?: Partial<OptionalProps>
): JSX.Element => {
  const handleSubmit = async (values) => {
    await sleep(500);
    optionalProps.onSubmit(values);
  };
  return (
    <ThemeProvider>
      <div>
        <Formik
          initialValues={{
            nestedInput: "initial value",
          }}
          onSubmit={handleSubmit}
        >
          <Form>
            <Field
              id="nestedInput"
              name="nestedInput"
              placeholder="Placeholder text"
              buttonText="Click Button"
              component={NestedInput}
              value="initial value"
              {...optionalProps}
            />
          </Form>
        </Formik>
      </div>
    </ThemeProvider>
  );
};

test("Standard NestedInput renders without crashing, matches snapshot", () => {
  const { container } = render(<NestedInputFormik />);
  expect(container).toMatchSnapshot();
});

test("NestedInput with usePhoneNumber renders without crashing, matches snapshot", () => {
  const { container } = render(<NestedInputFormik usePhoneNumber />);
  expect(container).toMatchSnapshot();
});

test("NestedInput with useVerificationCode renders without crashing, matches snapshot", () => {
  const { container } = render(<NestedInputFormik useVerificationCode />);
  expect(container).toMatchSnapshot();
});

test("Standard NestedInput props pass and render correctly", () => {
  const { getByPlaceholderText } = render(
    <NestedInputFormik useSubmit={false} />
  );

  const input = getByPlaceholderText("Placeholder text");
  expect(input).toBeTruthy();
  expect(input.getAttribute("id")).toEqual("nestedInput");
  expect(input.getAttribute("name")).toEqual("nestedInput");
  expect(input.getAttribute("value")).toBe("initial value");
  fireEvent.change(input, { target: { value: "eric@playpickup.com" } });
  expect(input.getAttribute("value")).toBe("eric@playpickup.com");
});

test("Phone NestedInput props pass and render correctly", () => {
  const { getByPlaceholderText, getByLabelText } = render(
    <NestedInputFormik usePhoneNumber label="Phone input" useSubmit={false} />
  );

  expect(getByLabelText("Phone input")).toBeTruthy();
  const input = getByPlaceholderText("(345) 555-2343");
  expect(input).toBeTruthy();
  expect(input.getAttribute("id")).toEqual("nestedInput");
  expect(input.getAttribute("name")).toEqual("nestedInput");
  expect(input.getAttribute("value")).toBe("");
  fireEvent.change(input, { target: { value: "6151234567" } });
  expect(input.getAttribute("value")).toBe("(615) 123-4567");
});

test("Verification code NestedInput props pass and render correctly", () => {
  const { getByPlaceholderText, getByLabelText } = render(
    <NestedInputFormik
      useVerificationCode
      label="Verification code"
      useSubmit={false}
    />
  );

  expect(getByLabelText("Verification code")).toBeTruthy();
  const input = getByPlaceholderText("Placeholder text");
  expect(input).toBeTruthy();
  expect(input.getAttribute("id")).toEqual("nestedInput");
  expect(input.getAttribute("name")).toEqual("nestedInput");
  expect(input.getAttribute("value")).toBe("initial value");
  fireEvent.change(input, { target: { value: "123456" } });
  expect(input.getAttribute("value")).toBe("123456");
});

test("Submit is passed when useSubmit is true", async () => {
  const handleSubmit = jest.fn();
  const { getByText } = render(<NestedInputFormik onSubmit={handleSubmit} />);
  const button = getByText("Click Button");
  expect(button).toBeTruthy();
  fireEvent.click(button);
  await waitFor(() =>
    expect(handleSubmit).toHaveBeenCalledWith({
      nestedInput: "initial value",
    })
  );
});

test("onClick is passed when useSubmit is false", () => {
  const onClick = jest.fn();
  const { getByText } = render(
    <NestedInputFormik useSubmit={false} onClick={onClick} />
  );
  const button = getByText("Click Button");
  expect(button).toBeTruthy();
  fireEvent.click(button);
  expect(onClick).toBeCalled();
});

test("Disabled prop passes and is rendered correctly", () => {
  const standardComponent = render(<NestedInputFormik disabled={true} />);
  const input = standardComponent.getByPlaceholderText("Placeholder text");
  const submitButton = standardComponent.getByText("Click Button");
  const phoneComponent = render(
    <NestedInputFormik usePhoneNumber disabled={true} useSubmit={false} />
  );
  const phoneInput = phoneComponent.getByPlaceholderText("(345) 555-2343");
  const standardButton = phoneComponent.getByTestId("pickup-nested-button");
  expect(input.hasAttribute("disabled"));
  expect(phoneInput.hasAttribute("disabled"));
  expect(submitButton.hasAttribute("disabled"));
  expect(standardButton.hasAttribute("disabled"));
});

test("breakpoint nested input is rendred correctly", () => {
  const { getByTestId } = render(<NestedInputFormik usePhoneNumber />);
  const hidden = getByTestId("breakpoint-large-div");
  const shown = getByTestId("breakpoint-small-div");

  const breakpointShownStyle = window.getComputedStyle(shown);
  expect(breakpointShownStyle.display).toBe("inline");
  const breakpointHiddenStyle = window.getComputedStyle(hidden);
  expect(breakpointHiddenStyle.display).toBe("none");
});
