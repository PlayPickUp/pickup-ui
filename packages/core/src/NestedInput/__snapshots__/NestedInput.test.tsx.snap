// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedInput with usePhoneNumber renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j8 {
  background: #7A60FF;
  color: #FFF;
  border: none;
  height: auto;
  margin: 0px 0px 0px 0px;
  display: flex;
  opacity: 1;
  padding: 0 30px;
  position: relative;
  flex-flow: row nowrap;
  font-size: 18px;
  appearance: none;
  background: linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%);
  min-height: 47px;
  text-align: center;
  transition: opacity 200ms ease-in-out;
  align-items: center;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 24px;
  border-radius: 4px;
  letter-spacing: 0.3px;
  text-transform: capitalize;
  justify-content: center;
}

body #PickUpUI .j9 {
  width: auto;
}

body #PickUpUI .j12 {
  width: 100%;
}

body #PickUpUI .j0 {
  width: 100%;
  display: flex;
  padding: 0 6px;
  position: relative;
  flex-flow: row nowrap;
  max-width: 360px;
  align-items: center;
  border-radius: 8px;
  justify-content: space-between;
  background-color: #FFF;
}

body #PickUpUI .j1 {
  border: 1px solid #DFDAE6;
  height: 60px;
}

body #PickUpUI .j2 {
  flex: 1 1 auto;
  padding-right: 8px;
}

body #PickUpUI .j3 {
  display: flex;
  position: relative;
  flex-flow: row nowrap;
  align-items: center;
}

body #PickUpUI .j4 {
  color: #615E66;
  padding: 0 8px;
  position: relative;
  font-size: 15px;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 22px;
}

body #PickUpUI .j5 {
  color: #615E66;
  width: 100%;
  border: none;
  font-size: 16px;
  appearance: none;
  font-family: 'Inter', Helvetica, sans-serif;
  letter-spacing: 0.1px;
}

body #PickUpUI .j6 {
  height: 22px;
  line-height: 22px;
}

body #PickUpUI .j7 {
  display: none;
}

body #PickUpUI .j10 {
  margin: 0;
  padding: 0 8px;
  font-size: 14px;
  font-weight: normal;
}

body #PickUpUI .j11 {
  display: inline;
}

<div>
  <div
    id="PickUpUI"
  >
    <div>
      <form
        action="#"
      >
        <div
          class="j0 j1"
        >
          <div
            class="j2"
          >
            <div
              class="j3"
            >
              <div
                class="j4"
              >
                +1
              </div>
              <input
                autocomplete="tel"
                class="j5 j6"
                id="nestedInput"
                name="nestedInput"
                placeholder="(345) 555-2343"
                type="tel"
                value=""
              />
            </div>
          </div>
          <div
            class="j7"
            data-testid="breakpoint-large-div"
          >
            <input
              class="j8 j9 j10"
              data-testid="pickup-button"
              style="min-width: 90px;"
              type="submit"
              value="Click Button"
            />
          </div>
        </div>
        <div
          class="j11"
          data-testid="breakpoint-small-div"
        >
          <input
            class="j8 j12 j10"
            data-testid="pickup-button"
            style="min-width: 90px; margin-top: 4px;"
            type="submit"
            value="Click Button"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`NestedInput with useVerificationCode renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j6 {
  background: #7A60FF;
  color: #FFF;
  border: none;
  height: auto;
  margin: 0px 0px 0px 0px;
  display: flex;
  opacity: 1;
  padding: 0 30px;
  position: relative;
  flex-flow: row nowrap;
  font-size: 18px;
  appearance: none;
  background: linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%);
  min-height: 47px;
  text-align: center;
  transition: opacity 200ms ease-in-out;
  align-items: center;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 24px;
  border-radius: 4px;
  letter-spacing: 0.3px;
  text-transform: capitalize;
  justify-content: center;
}

body #PickUpUI .j7 {
  width: auto;
}

body #PickUpUI .j10 {
  width: 100%;
}

body #PickUpUI .j0 {
  width: 100%;
  display: flex;
  padding: 0 6px;
  position: relative;
  flex-flow: row nowrap;
  max-width: 360px;
  align-items: center;
  border-radius: 8px;
  justify-content: space-between;
  background-color: #FFF;
}

body #PickUpUI .j1 {
  border: 1px solid #DFDAE6;
  height: 56px;
}

body #PickUpUI .j2 {
  flex: 1 1 auto;
  padding-right: 8px;
}

body #PickUpUI .j3 {
  color: #615E66;
  width: 100%;
  border: none;
  font-size: 16px;
  appearance: none;
  font-family: 'Inter', Helvetica, sans-serif;
  letter-spacing: 0.1px;
}

body #PickUpUI .j4 {
  height: 22px;
  line-height: 22px;
}

body #PickUpUI .j5 {
  display: none;
}

body #PickUpUI .j8 {
  margin: 0;
  padding: 0 8px;
  font-size: 14px;
  font-weight: normal;
}

body #PickUpUI .j9 {
  display: inline;
}

<div>
  <div
    id="PickUpUI"
  >
    <div>
      <form
        action="#"
      >
        <div
          class="j0 j1"
        >
          <div
            class="j2"
          >
            <input
              class="j3 j4"
              id="nestedInput"
              name="nestedInput"
              placeholder="Placeholder text"
              value="initial value"
            />
          </div>
          <div
            class="j5"
            data-testid="breakpoint-large-div"
          >
            <input
              class="j6 j7 j8"
              data-testid="pickup-button"
              style="min-width: 90px;"
              type="submit"
              value="Click Button"
            />
          </div>
        </div>
        <div
          class="j9"
          data-testid="breakpoint-small-div"
        >
          <input
            class="j6 j10 j8"
            data-testid="pickup-button"
            style="min-width: 90px; margin-top: 4px;"
            type="submit"
            value="Click Button"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Standard NestedInput renders without crashing, matches snapshot 1`] = `
body #PickUpUI .j6 {
  background: #7A60FF;
  color: #FFF;
  border: none;
  height: auto;
  margin: 0px 0px 0px 0px;
  display: flex;
  opacity: 1;
  padding: 0 30px;
  position: relative;
  flex-flow: row nowrap;
  font-size: 18px;
  appearance: none;
  background: linear-gradient(112.25deg, #914EFF -20.38%, #5D1BFF 122.56%);
  min-height: 47px;
  text-align: center;
  transition: opacity 200ms ease-in-out;
  align-items: center;
  font-family: 'Inter', Helvetica, sans-serif;
  line-height: 24px;
  border-radius: 4px;
  letter-spacing: 0.3px;
  text-transform: capitalize;
  justify-content: center;
}

body #PickUpUI .j7 {
  width: auto;
}

body #PickUpUI .j0 {
  width: 100%;
  display: flex;
  padding: 0 6px;
  position: relative;
  flex-flow: row nowrap;
  max-width: 360px;
  align-items: center;
  border-radius: 8px;
  justify-content: space-between;
  background-color: #FFF;
}

body #PickUpUI .j1 {
  border: 1px solid #DFDAE6;
  height: 56px;
}

body #PickUpUI .j2 {
  flex: 1 1 auto;
  padding-right: 8px;
}

body #PickUpUI .j3 {
  color: #615E66;
  width: 100%;
  border: none;
  font-size: 16px;
  appearance: none;
  font-family: 'Inter', Helvetica, sans-serif;
  letter-spacing: 0.1px;
}

body #PickUpUI .j4 {
  height: 48px;
  line-height: 24px;
}

body #PickUpUI .j5 {
  display: none;
}

<div>
  <div
    id="PickUpUI"
  >
    <div>
      <form
        action="#"
      >
        <div
          class="j0 j1"
        >
          <div
            class="j2"
          >
            <input
              class="j3 j4"
              id="nestedInput"
              name="nestedInput"
              placeholder="Placeholder text"
              value="initial value"
            />
          </div>
          <div
            class="j5"
            data-testid="breakpoint-large-div"
          >
            <input
              class="j6 j7"
              data-testid="pickup-button"
              style="min-width: 90px;"
              type="submit"
              value="Click Button"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
